contracts/gauges/Gauge.sol

```diff
@@ -53,7 +69,34 @@ contract Gauge is IGauge, ERC2771Context, ReentrancyGuard {
       return rewardPerTokenStored;
     }
     return
-      rewardPerTokenStored + ((lastTimeRewardApplicable() - lastUpdateTime) * rewardRate * PRECISION) / totalSupply;
+      rewardPerTokenStored + ((lastTimeRewardApplicable() - lastUpdateTime) * rewardRate * PRECISION) / gainTotalSupply;
+  }
+
+//  function rewardPerTokenWithShare(address _user) public view returns (uint256)  {
+//    uint256 _pertoken = rewardPerToken();
+//    uint256 _reward = _pertoken * wLP / Factor + _pertoken * wVote / Factor;
+////    _pertoken *(wlp/factor + wVote/factor)
+//    return _reward;
+//  }
+
+  function depositShare(address _user, uint256 _share) external {
+    if (msg.sender != voter) revert NotVoter();
+    uint256 _oldShare = shares[_user];
+    shares[_user] = _share;
+    totalShare = totalShare - _oldShare + _share;
+    _updateRewards(_user);
+    updateGainBalance(_user);
+  }
+
+  function updateGainBalance(address _user) internal {
+    uint256 _originBalance = balanceOf[_user];
+    if ( _originBalance > 0 ){
+      uint256 _share = shares[_user];
+      uint256 _gainBalance = _originBalance * gain / Factor * _share / gainTotalSupply + _originBalance;
+      uint256 _oldGainBalance = gainBalanceOf[_user];
+      gainBalanceOf[_user] = _gainBalance;
```

Comment:
`gainBalance = balance * (Factor * share / totalShare + 1)`
such that, `gainBalanceOf` can be deleted.

Reply:
same reason with above